src/domain/entities/figma-design-identifier.ts

```diff
@@ -40,12 +40,16 @@ export class FigmaDesignIdentifier {
 		);
 
 		const fileKey = pathComponents[filePathComponentId + 1];
+		const branchFileKey =
+			pathComponents[filePathComponentId + 2] === 'branch'
+				? pathComponents[filePathComponentId + 3]
+				: undefined;
 		const nodeId = url.searchParams.get('node-id')?.replace('-', ':');
 
 		if (!fileKey)
 			throw new Error(`Received invalid Figma URL: ${url.toString()}`);
 
-		return new FigmaDesignIdentifier(fileKey, nodeId);
+		return new FigmaDesignIdentifier(branchFileKey ?? fileKey, nodeId);
 	};
```

Comment:
**note:** There is also the code that creates a normalised Figma design URLs based on `FigmaDesignIdentifier`:

- https://github.com/atlassian-labs/figma-for-jira/blob/9a59d71c167999678fd0d125bdef9366db08941a/src/infrastructure/figma/transformers/utils.ts#L10
- https://github.com/atlassian-labs/figma-for-jira/blob/9a59d71c167999678fd0d125bdef9366db08941a/src/infrastructure/figma/transformers/utils.ts#L30C14-L30C29
- https://github.com/atlassian-labs/figma-for-jira/blob/9a59d71c167999678fd0d125bdef9366db08941a/src/infrastructure/figma/transformers/utils.ts#L53C14-L53C31

These functions are used to construct normalised URLs for certain `AtlassianDesign` fields, e.g.:

```ts
return {
		id: designId.toAtlassianDesignId(),
		// ...
		url: buildDesignUrl({ fileKey }).toString(),
		liveEmbedUrl: buildLiveEmbedUrl({ fileKey }).toString(),
		inspectUrl: buildInspectUrl({ fileKey }).toString(),
		// ...
	};
```

As a result, given `https://www.figma.com/file/my-file-key/branch/my-branch-file-key/my-branch-name`, the app will construct `AtlassianDesign` with branch-unaware URLs, e.g:

```ts
{
  id: "my-branch-file-key",
  // ...
  url: "https://www.figma.com/file/my-branch-file-key",
  liveEmbedUrl: "https://www.figma.com/file/my-branch-file-key?mode=dev",
	inspectUrl: "https://www.figma.com/embed?embed_host=figma-jira-add-on&url=https://www.figma.com/file/my-branch-file-key?mode=dev",
  // ...
}
```

From what I see from my testing, this is the correct behavior as Figma resolves branch-unaware URLs to full branch-aware ones. This is the area we need to test carefully.